### DISCUSSION TITLE
Command to center element on the mobile screen

### DISCUSSION THREAD

**Discussion by mnkravets** [appium/appium#15776] [General]
Hi everyone, 
Is there any command I can use to center desired element on the screen, using Appium with JavaScript? (to avoid swiping or scrolling)
Thank you.

**Comment by jlipps** [comment]
If you're using Appium with a web browser, then sure, you can use executeScript to run some JavaScript that sets `window.scrollTop` or whatever the appropriate variable is.

**Comment by mnkravets** [comment]
I'm using Android emulator. I tried to use this script (works fine for web pages), received an error : Method is not implemented.  

browser.execute(function (_xpath) {
let element = document.evaluate(_xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
element.scrollIntoView({ block: "center" });

The idea is to pass the xpath of the element and to center an element on the mobile screen.

**Reply by jlipps** [reply]
What is `browser.execute`? Where is it defined in the documentation for the client library you are using?

**Reply by jlipps** [reply]
And are you in a web browser or in a native app?

**Reply by mnkravets** [reply]
I'm trying to build a custom JS command to center an element on the mobile screen. I'm in native app. 
browser.execute that's from Webdriver.io API. 

(() => {
browser.addCommand("centerElement",function (xpath) {
this._validateParameters({ xpath: xpath });

 try {
      browser.execute(function (_xpath) {
      let element = document.evaluate(_xpath, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
                    element.scrollIntoView();
                }, xpath);
            } catch (ex) {
            //
            }
        },
    );
})();

browser.centerElement(xpath); - is this idea even possible? Or better to use scrolling and swiping in order to reach an element?

**Reply by jlipps** [reply]
Native apps don't support JavaScript. Only webpages. That's why execute script is not implemented. iOS and Android don't provide JavaScript APIs to their native code.

**Reply by mnkravets** [reply]
Got it, thank you for responses!
